packages/drivers/odsp-driver/src/getUrlAndHeadersWithAuth.ts

```diff
@@ -2,16 +2,15 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+import { assert } from "@fluidframework/core-utils";
 
 export function getUrlAndHeadersWithAuth(
 	url: string,
 	// eslint-disable-next-line @rushstack/no-new-null
-	token: string | null,
+	token: string,
 	forceAccessTokenViaAuthorizationHeader: boolean,
 ): { url: string; headers: { [index: string]: string } } {
-	if (!token || token.length === 0) {
-		return { url, headers: {} };
-	}
+	assert(token.length > 0, "should be token");
```

Comment:
I checked the usage of `getUrlAndHeadersWithAuth` in the code and this looks like a safe change, except there are a couple of places where we log telemetry as `headers:false` if authorization header was not present in the older code. I wonder if it makes sense to remove those logs as well. Don't have a strong opinion about this, not sure what was the original intent there.